components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/confirmation/ResendConfirmationManager.java

```diff
@@ -180,8 +182,14 @@ public ResendConfirmationDTO resendConfirmation(String tenantDomain, String rese
             confirmationCode = Utils.generateSecretKey(notificationChannel, user.getTenantDomain(), recoveryScenario);
             confirmationCode = Utils.concatRecoveryFlowIdWithSecretKey(recoveryFlowId, notificationChannel,
                     confirmationCode);
+            try {
+                hashedConfirmationCode = Utils.doHash(confirmationCode);
```

Comment:
The current `Utils.doHash` method throws UserStoreException if the algorithm is not available. IMO its not correct and hard to use. Shall we introduce a separate method to make it easy to read this code.

Reply:
Fixed with c5821975215ed0a9ab572c698e20b59881a214e9